Applied existing sanitization/validation rules

lecegues opened the pull request:
### Description
- Applied existing sanitization/validation middleware in backend
- Added to existing frontend validation in frontend to match backend rules
- Updated existing tests & added new test for changes

lecegues committed:
Applied sanitization/validation rules for users & manually tested (backend)

lecegues committed:
Updated some sanitization/validation rules

lecegues committed:
Applied validation & sanitization middleware to necessary routes

lecegues committed:
Added frontend validation rules derived from the backend validation rules

lecegues committed:
Updated existing tests and added a new test due to changes in code.

funmise reviewed (approved):
Good work but needs to be updated due to changes in projectile-point e.g. name field no longer projectile modal, now auto generated

lecegues committed:
Merge branch 'develop' into feature/validation

lecegues committed:
Fixed some unintended changes from merging/unhandled merge conflicts

lecegues committed:
Fixed linting errors that stemmed from unnoticed auto-merge conflicts

lecegues committed:
Fixed bugs relating to feature/projectile-point (out of scope), but necessary

lecegues committed:
Updated frontend & backend validation, especially for `users` username/password changes

lecegues committed:
Updated unit tests according to changes in code. Specifically:

1. Backend - Sanitization/Validation unit tests: Removed `registration` unit tests because it was replaced with `changeUsername/Password`
           - Added test suites for changeUsername and changePassword validation/sanitization rules

2. Frontend - Removed a test that checked for a JS Alert() - uses input validation rather than showing a JS alert
